### PR TITLE
[ASCII-976] split `pkg/status` into sub packages 

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Organize the `pkg/status` into separate subpackages.

This work is part of the ongoing work to enable the use of the new status component `comp/core/status`. Here is a POC that shows how we would use the status component in the future https://github.com/DataDog/datadog-agent/pull/21723

The work to separate functionality into subpackages would help us during the migration. 

### Describe how to test/QA your changes

The output of the status commands works as expected.

Normal agent

- `agent status`
- `agent launch-gui`

Cluster agent

- `agent status`

Process agent

- `/opt/datadog-agent/embedded/bin/process-agent status`

Security Agent

- `agent status`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/status/common/common.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package common provides a functoin to get the status elements common to all Agent flavors.
```
Comment (pgimalac) [inline pkg/status/common/common.go:6]:
```suggestion
// Package common provides a function to get the status elements common to all Agent flavors.
```

File: pkg/status/clusteragent/status_apiserver.go [hunk 2]
```diff
@@ -1,11 +1,11 @@
 // Unless explicitly stated otherwise all files in this repository are licensed
 // under the Apache License Version 2.0.
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
-// Copyright 2016-present Datadog, Inc.
+// Copyright 2023-present Datadog, Inc.
```
Comment (pgimalac) [inline pkg/status/clusteragent/status_apiserver.go:4]: Just to make sure: this is on purpose and the file wasn't created before 2023 ?
(same question for the other files were you changed the copyright year)

FYI 4200 Go files out of 4900 have a copyright with year 2016 so I don't think it's a big deal to leave 2016 :smile:

File: pkg/status/endpoints/endpoints.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+// Package endpoints fetch information needed to render the 'endpoints' section of the status page.
+package endpoints
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/utils"
+)
+
+// PopulateStatus populates the status stats
+func PopulateStatus(stats map[string]interface{}) {
```
Comment (pgimalac) [inline pkg/status/endpoints/endpoints.go:15]: **💭 thought:** ‏You could have kept `getEndpointsInfos` and just have `PopulateStatus` check the return value and error

**Review by pgimalac — APPROVED** [review]
LGTM !

I think some files should be owned or co-owned by different teams, eg `pkg/status/cluster-agent`, `pkg/status/dogstatsd`, etc

**Comment by GustavoCaso** [comment]
> I think some files should be owned or co-owned by different teams, eg pkg/status/cluster-agent, pkg/status/dogstatsd, etc

This is an intermediate state. Once we migrate those bits to their component the codeowner would be the right team.

**Review by AliDatadog — APPROVED** [review]
/LGTM for our owned files. Please make sure that the status command still works on the Cluster-Agent as well as part of the QA.

**Pull request merged**